### DISCUSSION TITLE
Uncaught (in promise) TypeError: Cannot read properties of null (reading 'fingerprint')

### DISCUSSION THREAD

**Discussion by ItsJoshDaly** [livewire/livewire#3712] [1. Help] [answered]
Hi there,

We are building a dynamic form with livewire components. We use a select element with a given ID to decide which livewire components we want to render to build the form. We can refresh our components up to three times before we come across an error.

Uncaught (in promise) TypeError: Cannot read properties of null (reading 'fingerprint')

Unfortunately, even following the advice of the troubleshooting docs, which includes wrapping the conditionals in divs and applying the IDs to divs, and even calling time() has not helped and the issue still remains. Here is the code block.

`@foreach($components as $component)
                            @if($component->component_type == "supplemented_dropdown_input")
                                <livewire:suppdropdown :component="$component" :wire:key="time().$component->id" />
                            @elseif($component->component_type == "checkbox_input")
                                <livewire:checkboxinput :component="$component" />
                            @elseif($component->component_type == "text_input")
                                <livewire:text-input :component="$component" />
                            @elseif($component->component_type == "image_input")
                                <livewire:choosephoto :component="$component" />
                            @endif
                        @endforeach`

I have attempted numerous times to DD after the error appears, but we get no information from that either. Any ideas on how to fix this? Thanks very much.

**Comment by joshhanley** [comment, marked as answer]
@ItsJoshDaly can you please make sure to format your code correctly as it is hard to read 🙂

When you paste code into github, you need to surround it in triple back ticks and then specify the file type like php or blade.

For example
\```php
\\ PHP code here
\```

\```blade
\\ HTML and blade code here
\```

The issue will be the fact that the` $component` is used internally, so you are probably running into collisions. So it's probably actually using the Livewire component id, which changes each time so, using it as part of `wire:key` will be causing problems.

You need to try to come up with a key that is unique but consistent between requests. So using `time()` isn't a good idea as it will for all children to be removed and re-added on every request.

I would try renaming `$component` to something else like `$element` for form element, removing `time()` from the key, and also adding some context to the key so it's not just an ID such as `wire:key="dynamic-element-{{ $element->id }}"`.

Hope this helps!
